client/post-editor/media-modal/detail/test/index.jsx

```diff
@@ -66,26 +77,20 @@ describe( 'EditorMediaModalDetailItem', () => {
 			/>
 		);
 
-		const editButton = tree.find( '.editor-media-modal-detail__edit' );
-
-		expect( editButton.length ).toBeGreaterThanOrEqual( 1 );
+		expect( screen.queryAllByRole( 'button' ).length ).toBeGreaterThanOrEqual( 1 );
 	} );
 
 	test( 'should display at least one edit button for an image on a public site', () => {
-		const tree = shallow( <DetailItem item={ DUMMY_IMAGE_MEDIA } { ...SHARED_PROPS } /> );
-
-		const editButton = tree.find( '.editor-media-modal-detail__edit' );
+		renderWithRedux( <DetailItem item={ DUMMY_IMAGE_MEDIA } { ...SHARED_PROPS } /> );
 
-		expect( editButton.length ).toBeGreaterThanOrEqual( 1 );
+		expect( screen.queryAllByRole( 'button' ).length ).toBeGreaterThanOrEqual( 1 );
 	} );
 
 	test( 'should not display edit button for an image on a private site', () => {
-		const tree = shallow(
+		renderWithRedux(
 			<DetailItem item={ DUMMY_IMAGE_MEDIA } isSitePrivate={ true } { ...SHARED_PROPS } />
 		);
 
-		const editButton = tree.find( '.editor-media-modal-detail__edit' );
-
-		expect( editButton ).toHaveLength( 0 );
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
```

Comment:
The previous test asserted that we don't have any edit buttons, but theoretically, if we had other buttons, it would pass. That's not the case for the new version of the test. This should be solvable with Markus' suggestion above of making the button query a bit more specific.